ldap_auth_provider.py

```diff
@@ -41,11 +45,39 @@ class LDAPMode(object):
     LIST = (SIMPLE, SEARCH)
 
 
-class LdapAuthProvider(object):
+SUPPORTED_LOGIN_TYPE = "m.login.password"
+SUPPORTED_LOGIN_FIELDS = ('password',)
+
+
+class LdapAuthProvider:
     _ldap_tls = ldap3.Tls(validate=ssl.CERT_REQUIRED)
 
-    def __init__(self, config, account_handler):
-        self.account_handler = account_handler
+    def __init__(self, config, account_handler=None, api: Optional["ModuleApi"] = None):
+        if api is not None:
```

Comment:
wouldn't it be better to have two different classes, than try to figure out what whether we're being used with the old or the new API?